Flaky test: TestSplitRangeForTable in tests/realtikvtest/importintotest4

Automated flaky test report.

Title: Flaky test: TestSplitRangeForTable in tests/realtikvtest/importintotest4
- Repo: pingcap/tidb
- Package: tests/realtikvtest/importintotest4
- Case: TestSplitRangeForTable
- Branch: master
- Flaky Count: 2
- Time Thresholded Count: 0
- Latest Build: https://prow.tidb.net/jenkins/job/pingcap/job/tidb/job/pull_integration_realcluster_test_next_gen/2353/

Window: 2026-05-08T00:00:00.000Z → 2026-05-09T00:00:00.000Z
Threshold: 100000 ms

Note: This issue is created for a specific branch; updates may be posted as comments.

I re-checked the reopen evidence for this issue and I cannot confirm from build 4879 that `TestSplitRangeForTable` actually failed.

In build 4879:
- `bazel-go-test-problem-cases.json` marks `TestSplitRangeForTable` as `new_flaky`, but gives no concrete reason.
- `bazel-go-test-index.log` shows the case was run.
- The raw `bazel-target-output-L*` logs only show:
  - `=== RUN   TestImportInto/TestSplitRangeForTable`
  - `global_sort_test.go:405: In next-gen scenario we don't need 'force_partition_range' to import data`
  - `--- SKIP: TestImportInto/TestSplitRangeForTable (0.00s)`

I could not find any `--- FAIL: TestImportInto/TestSplitRangeForTable` in the raw logs.

At the same time, the same build does contain a direct failure for `TestGlobalSortRecordedStepSummary`, with an assertion failure at `tests/realtikvtest/importintotest4/global_sort_test.go:349`.

So from build 4879 alone, the reopen attribution to `TestSplitRangeForTable` does not look supported by the raw evidence. If there is another artifact showing an actual failure of this case, please point to it.

--- a/scripts/plugins/analyze-go-test-from-bazel-output.sh
+++ b/scripts/plugins/analyze-go-test-from-bazel-output.sh
@@ -90,11 +90,22 @@ function parse_bazel_go_test_new_flaky_cases() {
             for n in $targetShardOutputLineNum; do
                 local newFlakyCases=($(
                     sed -nE "/^${n}:/,/^[0-9]+:=+ Test output for \//p" "$DEFAULT_GO_TEST_INDEX_FILE" |
-                        grep -E "=== RUN|--- PASS" |
+                        grep -E "=== RUN|--- (PASS|FAIL|SKIP)" |
                         grep -Eo "\bTest\w+" | sort | uniq -c |
                         grep "^\s*1\b" |
                         grep -Eo "\bTest\w+"
                 ))
+
+                # Verify each candidate actually has a --- FAIL line in the index
+                local verifiedCases=()
+                for c in "${newFlakyCases[@]}"; do
+                    if grep -qE "--- FAIL:\s*${c}\b" "$DEFAULT_GO_TEST_INDEX_FILE"; then
+                        verifiedCases+=("$c")
+                    else
+                        echo "Skipping '${c}': no --- FAIL evidence in index"
+                    fi
+                done
+                newFlakyCases=("${verifiedCases[@]}")
 
                 ##### add into result json file.
                 if [ "${#newFlakyCases[@]}" -gt 0 ]; then

